src/corecel/cont/EnumBitset.hh

```diff
@@ -37,10 +37,11 @@ namespace celeritas
  * - hash support
  * - construct from string, from_ullong
  */
-template<size_type N>
-class Bitset
+template<class E>
+class EnumBitset
 {
-    static_assert(N > 0, "Zero-sized bitset is not supported");
+    static_assert(std::is_enum<E>::value, "Template parameter must be an enum");
```

Comment:
```suggestion
    static_assert(std::is_enum_v<E>, "Template parameter must be an enum");
    static_assert(sizeof(std::underlying_type_t<E>) <= sizeof(size_type), "Size of enum's underlying type must be smaller than size_type");
```

Reply:
Since the `size_` member has to fit inside the type, and our enum requires 2^size_ bits, I can't imagine we'd ever be able to define an enum with 2B values 😂

Reply:
haha I'd be surprised 😄  but if someone defines
 ```c++
enum class MyEnum : std::uint64_t
{
my_value = 64,
size_
}
```
they might run into issues with narrowing conversion when accessing that 64th bit since we `static_cast<size_type>`

Reply:
Sorry, I got confused 😅 , what we actually want is to avoid truncating the enum value when casting to `size_t`, so my original assertion is right. As Seth said, that would likely never happen